playback/tapeRecorder.ts

```diff
@@ -137,92 +208,263 @@ export class TapeRecorder {
+            : args
+        );
       } catch (error) {
         const errorMessage = `Input interception key creation error for alias \'${
           params.alias
         }\' - ${JSON.stringify(error)}`;
 
         if (that.inPlaybackMode()) {
-          throw new InputInterceptionKeyCreationError(errorMessage);
+          throw generatePlaybackException(
+            PlaybackExceptionTypes.InputInterceptionKeyCreationError,
+            errorMessage
+          );
         }
 
         console.error(errorMessage);
 
         interceptionKey = undefined;
         that.discardRecording();
       }
+      //#endregion
+
+      if (that.inPlaybackMode()) {
+        try {
+          return that.playbackRecordedInterception(
+            interceptionKey as string,
+            args
+          );
+        } catch (error) {
+          if (error.name == PlaybackExceptionTypes.RecordingKeyError) {
+            return params.func(...args);
+          }
+          throw error;
+        }
+      } else {
+        return that.executeFuncAndRecordInterception(
+          params.func,
+          args,
+          interceptionKey
+        );
+      }
+    }
+
+    return wrappedFunc;
+  }
+
+  /**
+   * Wrap a function that that acts as an output of the operation, the arguments are recorded as the output and
+   *    the result of the function is captured.
+   * output and the result of the function is captured
+   * @param params.alias - the alias of the function that was invoked
+   * @param params.func - the function to intercept
+   * @returns the intercepted function
+   */
+  public interceptOutput<inputType extends Array<any>, outputType>(params: {
+    alias: string;
+    func: (...args: inputType) => outputType;
+  }): (...args: inputType) => outputType {
+    const that = this;
+
+    function wrappedFunc(...args: inputType) {
+      if (!that.shouldIntercept()) {
+        return params.func(...args);
+      }
+
+      // If same alias(function) is invoked more than once we want to track each output invocation
+      that.invokeCounter[params.alias] =
+        that.invokeCounter[params.alias] + 1 || 1;
+      const invocationNumber = that.invokeCounter[params.alias] as number;
+
+      // Both in recording and playback mode we record what is sent to the output
+      that.recordOutput({ alias: params.alias, invocationNumber, args });
+
+      // Record output may have failed and discarded current recording which would make should intercept false
+      if (!that.shouldIntercept) {
+        return params.func(...args);
+      }
+
+      const interceptionKey =
+        that.outputInterceptionKey(params.alias, invocationNumber) + ".result";
 
       if (that.inPlaybackMode()) {
+        // Return recording of input invocation
         try {
-          return that.pla;
-        } catch (error) {}
+          return that.playbackRecordedInterception(interceptionKey, args);
+        } catch (error) {
```

Comment:
Looks like redundant try catch (copy paste from port but we still didn't implement the catch configurable handling)